assets/js/modules/analytics/components/settings/GA4SettingsControls.js

```diff
@@ -20,58 +20,50 @@
  * External dependencies
  */
 import PropTypes from 'prop-types';
-import { useUpdateEffect } from 'react-use';
 
 /**
  * WordPress dependencies
  */
-import { __, _x, sprintf } from '@wordpress/i18n';
-import { Fragment, useState, useCallback, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
 import { Button } from 'googlesitekit-components';
 import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
-import {
-	MODULES_ANALYTICS_4,
-	PROPERTY_CREATE,
-	WEBDATASTREAM_CREATE,
-} from '../../../analytics-4/datastore/constants';
+import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
 import { FORM_SETUP, MODULES_ANALYTICS } from '../../datastore/constants';
 import { CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
-import { Select, Option } from '../../../../material-components';
 import { GA4ActivateSwitch } from '../common';
-import { PropertySelect } from '../../../analytics-4/components/common';
+import {
+	PropertySelect,
+	MeasurementSelect,
+} from '../../../analytics-4/components/common';
 import SettingsUseSnippetSwitch from '../../../analytics-4/components/settings/SettingsUseSnippetSwitch';
 import JoyrideTooltip from '../../../../components/JoyrideTooltip';
 import GA4SettingsNotice from './GA4SettingsNotice';
 import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
 const { useSelect, useDispatch } = Data;
 
-export default function GA4SettingsControls( {
-	hasAnalyticsAccess,
-	hasAnalytics4Access,
-} ) {
-	const [ matchedProperty, setMatchedProperty ] = useState();
-	const [ matchedWebDataStream, setMatchedWebDataStream ] = useState();
+export default function GA4SettingsControls( props ) {
+	const { hasAnalyticsAccess, hasAnalytics4Access } = props;
+
+	const { setValues } = useDispatch( CORE_FORMS );
+	const { matchAndSelectProperty } = useDispatch( MODULES_ANALYTICS_4 );
 
 	const accountID = useSelect( ( select ) =>
 		select( MODULES_ANALYTICS ).getAccountID()
 	);
 
 	// This select is needed to check whether the AdminAPI works or not.
```

Comment:
This comment is no longer relevant as all "API works" logic has been removed.

Reply:
Removed.